src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,117 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double MULTIPLICATION_CAPACITY = 1.5;
+    private static final String EXCEPTION_MESSAGE = "Index of the bounds. Check your input index";
+    private T[] values;
+    private int size;
+
+    public ArrayList() {
+        size = 0;
+        values = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        grow();
+        values[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        indexOnRange(index);
+        grow();
+        for (int i = size; i > index; i--) {
+            values[i] = values[i - 1];
+        }
+        values[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        indexOnRangeWithEqualsSize(index);
+        return values[index];
     }
 
     @Override
     public void set(T value, int index) {
+        indexOnRangeWithEqualsSize(index);
+        T oldValue = values[index];
+        values[index] = value;
+    }
 
+    @Override
+    public T remove(T value) {
+        for (int i = 0; i < size; i++) {
+            if (values[i] == null) {
+                if (value == null) {
+                    T removed = values[i];
+                    remove(i);
+                    return removed;
+                }
+            } else {
+                if (values[i].equals(value)) {
+                    T removed = values[i];
+                    remove(i);
+                    return removed;
+                }
+            }
```

Comment:
No need to duplicate removing logic for `null` case, just add second condition to your `if (values[i].equals(value))`

Reply:
Fix maven build
<img width="307" alt="image" src="https://user-images.githubusercontent.com/72033531/220409283-def476b0-8471-484f-a601-254cc849da92.png">